### ISSUE TITLE
Persist Data

### ISSUE THREAD

**Issue by albertoammar** [Rocketseat/unform#474]
**What would you like to be added**:

Data persistence for hidden fields.

**Why is this needed**:

Sometimes selected fields can be temporarily removed, as an example of tabs, steps or filters on items.

**Additional context**

As an example below:

Each checkbox of this list has an "item.${id}", if I select the item and give the search when it goes to the original context, the item will no longer be selected, because for a while those items didn't exist.

![image](https://user-images.githubusercontent.com/14295479/181937128-11a12ae4-c67b-4301-a6ce-283c24b415f6.png)

**Comment by mrlonmra** [comment]
Boa observação! 
Good Observation!

**Comment by lucaswitch** [comment]
This is helpful for modals and tab navigations, very common.
Very needed feature.

**Comment by guilhermekodenvis** [comment]
Yep. I see this error while navigating between tabs

**Comment by lucaswitch** [comment]
For solving this issue maybe you should save the form data between tabs on a useState prop or useRef for avoiding re-renders.

**Comment by guilhermekodenvis** [comment]
Guys, are you still giving support for unform?

**Comment by albertoammar** [comment]
> 

Yes, I never said there's no way to do it. But the point of the library is not to have to do these hacks, right?

**Comment by albertoammar** [comment]
> For solving this issue maybe you should save the form data between tabs on a useState prop or useRef for avoiding re-renders.

Yes, I never said there's no way to do it. But the point of the library is not to have to do these hacks, right?
